physlr/physlr.py

```diff
@@ -400,14 +400,50 @@ def diameter_of_tree(g, weight=None):
         return (u, v, diameter)
 
     @staticmethod
-    def determine_backbones_of_trees(g):
-        "Determine the backbones of the maximum spanning trees."
+    def detect_junctions_of_tree(gcomponent, messages, min_branch):
+        """"
+        Detect the junctions in the tree, with >= 3 branches larger than min_branch.
+        """
+        candidate_junctions = {node
+                               for node in list(gcomponent.nodes)
+                               if gcomponent.degree(node) >= 3}
+        junctions = []
+        for candidate_junction in candidate_junctions:
+            candidate_messages = [messages[(candidate_junction, neighbor)]
+                                  for neighbor in gcomponent.neighbors(candidate_junction)]
+            candidate_messages.sort()
+            if candidate_messages[-3] >= min_branch:
+                # If the 3rd largest branch is considerably large, this node is a junction.
+                junctions.append(candidate_junction)
+        return junctions
+
+    @staticmethod
+    def remove_junctions_of_tree(min_branch, gcomponent):
+        """"
+        Detect and remove junctions of trees, with >= 3 branches larger than min_branch.
+        """
+        if min_branch == 0:
+            return gcomponent
+        messages = Physlr.determine_reachability_by_message_passing(gcomponent)
+        nodes_to_remove = \
+            Physlr.detect_junctions_of_tree(gcomponent, messages, min_branch)
```

Comment:
Modify `detect_junctions_of_tree` to call `determine_reachability_by_message_passing` rather than passing `messages`.